CreeDictionary/API/search/query.py

```diff
@@ -0,0 +1,46 @@
+import unicodedata
+from typing import Optional
+
+import marshmallow.fields
+
+from API.search.util import to_sro_circumflex
+
+
+class Query:
```

Comment:
I like this class!

One note: `SearchRun.__str__` delegates to the `__str__` of this class (which delegates to the default `__repr__`). Should this be given a simple `__repr__`. Something like?

```python
    def __repr__(self) -> str:
        return f"<Query {self._raw_query_string!r}>"
```